scripts/pc2_interactive.sh

```diff
@@ -108,11 +108,13 @@ KillValidator()
 HandleTerminateFromPC2()
 {
 	DEBUG echo "Received TERMINATE signal from PC2"
+	DEBUG echo "Killing off submission process group $submissionpid and all children"
 	KillValidator
-	if test -n "$submissionpid" -a -d /proc/$submissionpid
+	# Kill off all our kids
+	pkill -9 -P $$
```

Comment:
I am confused by the -P 
```
      -P, --parent ppid,...
              Only match processes whose parent process ID is listed.
```

If a process forks does it's parent process ID not get request to 1?

Reply:
that kills all processes that has the shell script pid ($$) as it's parent, but, doesnt kill the killer.
If it forks, it does not change its parent.  If it does a setsid() then yes, it becomes one, in which case this pkill would be a no-op.